### PR TITLE
[OPMONDEV-176]: move statistics collection logic from opendata to anonymizer module

### PR DESCRIPTION
**Pull request by wisecrow**
https://nordic-institute.atlassian.net/browse/OPMONDEV-176

Due to https://github.com/nordic-institute/X-Road-Metrics/blob/develop/docs/system_architecture.md#recommended-infrastructure-for-production-use
Opendata module cannot use MongoDB:

- [x] Move statistics collection from Opendata module to Anonymizer module
- [x]  Use `clean_data` instead of `raw_messages` to calculate statistics.
- [x] Store calculated statistics in PostgreSQL instead of MongoDB

### CODE REVIEW

File: anonymizer_module/metrics_statistics/postgresql_manager.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+        rows = []
+
+        record_values = [data.get(field_name) for field_name in self._field_order]
+        rows.append(cursor.mogrify(template, record_values).decode('utf-8'))
+
+        query_rows = ','.join(rows)
+        return f'INSERT INTO {self._table_name} ({column_names}) VALUES {query_rows}'
+
+    def is_alive(self):
```
Comment (justinask7) [inline anonymizer_module/metrics_statistics/postgresql_manager.py:130]: type hints would be nice

File: anonymizer_module/metrics_statistics/tests/test_statistics_manager.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import logging
+import sqlite3
+from datetime import datetime
```
Comment (justinask7) [inline anonymizer_module/metrics_statistics/tests/test_statistics_manager.py:3]: import order

File: anonymizer_module/metrics_statistics/tests/test_statistics_manager.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    },
+    'mongodb': {
+        'user': 'test-user',
+        'password': 'test-password',
+        'host': 'test host'
+    }
+}
+
+
+class MockPsyContextManager(object):
```
Comment (justinask7) [inline anonymizer_module/metrics_statistics/tests/test_statistics_manager.py:37]:
```suggestion
class MockPsyContextManager:
```

**Review by justinask7 — APPROVED** [review]

File: anonymizer_module/metrics_statistics/postgresql_manager.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            self._ensure_table(table_schema, index_columns)
+            self._ensure_privileges()
+
+    @staticmethod
+    def get_schema() -> List[Tuple[str, str]]:
+        schema = [
+            (field_name, field_data['type'])
+            for field_name, field_data in METRICS_STATISTICS_SCHEMA.items()
+        ]
+        schema.sort()
```
[the remark this answers is not in the document]
Comment (wisecrow) [reply anonymizer_module/metrics_statistics/postgresql_manager.py:96]: It sorts by `field_name` is this situation

**Pull request merged**